### PR TITLE
Extract bitmex-price-feed into dedicated actor

### PR DESCRIPTION
**Pull request by thomaseizinger**
This allows us to reuse this component in other codebases.

### CODE REVIEW

File: daemon/src/bitmex_price_feed.rs [hunk 1]
```diff
@@ -9,6 +8,7 @@ use rust_decimal::Decimal;
 use std::convert::TryFrom;
 use std::time::Duration;
 use time::OffsetDateTime;
+use tokio_tasks::Tasks;
```
Comment (luckysori) [inline daemon/src/bitmex_price_feed.rs:9]: 👍 Interesting that these use statements are available from `crate`. There are a lot of `use crate::Tasks` in this crate :(

File: daemon/src/supervisor.rs [hunk 2]
```diff
@@ -1,261 +0,0 @@
-use crate::Tasks;
```
Comment (luckysori) [inline daemon/src/supervisor.rs]: Ugh, thanks for finding this. The rebase must have reintroduced it.

File: daemon/src/projection.rs [hunk 3]
```diff
@@ -484,7 +484,7 @@ impl Cfd {
         let latest_quote = match latest_quote {
             Some(latest_quote) => latest_quote,
             None => {
-                tracing::debug!(order_id = %self.order_id, "Unable to calculate profit/loss without current price");
+                tracing::trace!(order_id = %self.order_id, "Unable to calculate profit/loss without current price");
```
Comment (luckysori) [inline daemon/src/projection.rs:499]: 🤩

File: daemon/src/projection.rs [hunk 4]
```diff
@@ -962,6 +975,12 @@ mod round_to_two_dp {
         }
     }
 
+    impl ToDecimal for Decimal {
```
Comment (luckysori) [inline daemon/src/projection.rs:995]: 😄

File: daemon/src/projection.rs [hunk 5]
```diff
@@ -502,7 +501,9 @@ impl Cfd {
         let latest_price = match Price::new(latest_price) {
             Ok(latest_price) => latest_price,
             Err(e) => {
-                tracing::warn!("Failed to compute profit/loss because latest price is invalid: {e}");
+                tracing::warn!(
+                    "Failed to compute profit/loss because latest price is invalid: {e}"
+                );
```
Comment (luckysori) [inline daemon/src/projection.rs:519]: ❓ Why do these diffs keep appearing? As in nothing has changed, but these lines are being reformatted.

**Review by luckysori — APPROVED** [review]
LGTM

File: daemon/src/projection.rs (hunk 5 above)
Comment (thomaseizinger) [reply daemon/src/projection.rs:519]: I don't know but I will run the formatter again on each commit to check!

File: daemon/src/bitmex_price_feed.rs (hunk 1 above)
Comment (thomaseizinger) [reply daemon/src/bitmex_price_feed.rs:9]: I'll see to pack them all in this commit!

Comment (luckysori) [reply daemon/src/bitmex_price_feed.rs:9]: Thanks!

File: daemon/src/projection.rs (hunk 4 above)
Comment (thomaseizinger) [reply daemon/src/projection.rs:995]: Needed that to make the serde module work :D

**Pull request merged**